src/status.rs

```diff
@@ -208,18 +208,21 @@ impl Serialize for Status {
         if self.cache.is_some() { field_count += 1; }
         if self.sensors.is_some() { field_count += 1; }
         if self.ext_versions.is_some() { field_count += 1; }
+        field_count += self.extensions.len();
 
         // Serialize fields
-        let mut state = serializer.serialize_struct("Status", field_count)?;
+        let mut state = serializer.serialize_map(Some(field_count))?;
```

Comment:
See https://github.com/serde-rs/serde/issues/708 and https://github.com/serde-rs/serde/issues/639